src/apps/filters/h3.c

```diff
@@ -41,25 +43,66 @@
+    }                                                                     \
+    }                                                                     \
+    ;                                                                     \
+                                                                          \
+    H3Error generalHelp(int argc, char *argv[]) {                         \
+        int arglen = sizeof(subcommands) / sizeof(subcommands[0]) - 1;    \
+        Arg **args = calloc(arglen, sizeof(Arg *));                       \
+        args[0] = &helpArg;                                               \
+        for (int i = 0; i < arglen - 1; i++) {                            \
+            args[i + 1] = subcommands[i].arg;                             \
+        }                                                                 \
+                                                                          \
+        const char *helpText =                                            \
+            "Please use one of the subcommands listed to perform an H3 "  \
+            "calculation. Use h3 <SUBCOMMAND> --help for details on the " \
+            "usage of "                                                   \
+            "any subcommand.";                                            \
+        if (parseArgs(argc, argv, arglen, args, &helpArg, helpText)) {    \
+            free(args);                                                   \
+            return E_SUCCESS;                                             \
+        } else {                                                          \
+            free(args);                                                   \
+            return E_FAILED;                                              \
+        }                                                                 \
+    }
+
+#define DISPATCH_SUBCOMMAND()                                                \
+    for (int i = 0; i < sizeof(subcommands) / sizeof(subcommands[0]); i++) { \
+        if (has(subcommands[i].name, 1, argv)) {                             \
+            return subcommands[i].subcommand(argc, argv);                    \
+        }                                                                    \
+    }
+
 bool has(char *subcommand, int level, char *argv[]) {
     return strcasecmp(subcommand, argv[level]) == 0;
 }
 
 Arg helpArg = ARG_HELP;
-Arg cellToLatLngArg = {
-    .names = {"cellToLatLng"},
-    .helpText = "Convert an H3 cell to a WKT POINT coordinate",
-};
-Arg latLngToCellArg = {
-    .names = {"latLngToCell"},
-    .helpText = "Convert degrees latitude/longitude coordinate to an H3 cell.",
-};
-Arg cellToBoundaryArg = {
-    .names = {"cellToBoundary"},
-    .helpText = "Convert an H3 cell to a WKT POLYGON defining its boundary",
-};
 
-H3Error cellToLatLngCmd(int argc, char *argv[]) {
+/// Indexing subcommands
+
+SUBCOMMAND(cellToLatLng, "Convert an H3Cell to a WKT POINT coordinate") {
```

Comment:
Maybe it is too late to change at this point, but it seems unintuitive that the function ends with `LatLng` but it returns a WKT point, which is in Lng/Lat order.

Is it too late to call this function `cellToWKTPoint`? Or to change the output to be a lat/lng pair?

Reply:
@isaacbrodsky and I discussed this in the last PR. We're going to try and come up with a few "standard" input and output formats where applicable and have as much overlap between subcommands as possible.

But for this first pass of just getting an implementation actually done, since this is just happening when I have some spare time, it's whichever we think is the "most useful" standard, and Isaac asked for this to output WKT. I originally had it as a bare `lat, lng`.

So, I wouldn't rename them, as I'd rather the names match the h3geo.org documentation. Having it output in the ordering you mention will come. And I wouldn't mind if others wanted to chip in. ;)

Reply:
I remember the previous discussion, but I guess I'd really like a `--wkt` or `--format=wkt` flag here, just because otherwise we have an API we can't change in a backwards-compatible way...